src/ILLink.RoslynAnalyzer/RequiresAnalyzerBase.cs

```diff
@@ -0,0 +1,249 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace ILLink.RoslynAnalyzer
+{
+	[DiagnosticAnalyzer (LanguageNames.CSharp)]
+	public abstract class RequiresAnalyzerBase : DiagnosticAnalyzer
+	{
+		private protected abstract string RequiresAttributeName { get; }
+
+		private protected abstract string RequiresAttributeFullyQualifiedName { get; }
+
+		private protected abstract DiagnosticTargets AnalyzerDiagnosticTargets { get; }
+
+		private protected abstract DiagnosticDescriptor RequiresDiagnosticRule { get; }
+
+		public override void Initialize (AnalysisContext context)
+		{
+			context.EnableConcurrentExecution ();
+			context.ConfigureGeneratedCodeAnalysis (GeneratedCodeAnalysisFlags.ReportDiagnostics);
+			context.RegisterCompilationStartAction (context => {
+				var compilation = context.Compilation;
+				if (VerifyMSBuildOptions (context.Options, compilation) == false)
```

Comment:
`== false`

Reply:
Maybe it would be better if this were called `IsAnalyzerEnabled`? If a function returns a bool the standard C# convention is to use a prefix like `Is` or `Has`. `Verify` sounds more like a `void` function that performs some sort of side effect.